libraries/rush-lib/src/logic/operations/DependencyAnalysisPlugin.ts

```diff
@@ -0,0 +1,330 @@
+  dependencies: string[];
+  dependents: string[];
+}
+
+export type IDependencyGraph = Record<string, IDependencyGraphEntry>;
+
+interface IEnsureAndGetDependencyGraphEntryOptions {
+  packageName: string;
+  dependencyGraph: IDependencyGraph;
+}
+
+interface IAddDependenciesOptions {
+  dependencies: IPackageJsonDependencyTable | undefined;
+  dependencyGraph: IDependencyGraph;
+  project: RushConfigurationProject;
+  rushConfiguration: RushConfiguration;
+}
+
+interface IProcessPackageOptions {
+  dependencyGraph: IDependencyGraph;
+  project: RushConfigurationProject;
+  rushConfiguration: RushConfiguration;
+}
+
+interface IDependencyEntry {
+  project: string;
+  dependencies: string[];
+  dependents: string[];
+  buildTime: number;
+}
+
+export class DependencyAnalysisPlugin implements IPhasedCommandPlugin {
+  private readonly _terminal: ITerminal;
+
+  public constructor(terminal: ITerminal) {
+    this._terminal = terminal;
+  }
+
+  public apply(hooks: PhasedCommandHooks): void {
+    hooks.afterExecuteOperations.tap(
+      PLUGIN_NAME,
+      (result: IExecutionResult, context: ICreateOperationsContext): void => {
+        _printDependencyAnalysis(this._terminal, result);
+      }
+    );
+  }
+}
+
+function ensureAndGetDependencyGraphEntry({
+  packageName,
+  dependencyGraph
+}: IEnsureAndGetDependencyGraphEntryOptions): IDependencyGraphEntry {
+  if (dependencyGraph[packageName] === undefined) {
+    dependencyGraph[packageName] = { dependencies: [], dependents: [] };
+  }
+
+  return dependencyGraph[packageName];
+}
+
+function addDependencies({
+  dependencies = {},
+  dependencyGraph,
+  project,
+  rushConfiguration
+}: IAddDependenciesOptions): void {
+  const dependencyGraphEntry: IDependencyGraphEntry = ensureAndGetDependencyGraphEntry({
+    dependencyGraph,
+    packageName: project.packageName
+  });
+  Object.keys(dependencies).forEach((dependencyName) => {
+    if (
+      rushConfiguration.projectsByName.has(dependencyName) &&
+      !project.cyclicDependencyProjects.has(dependencyName)
+    ) {
+      dependencyGraphEntry.dependencies.push(dependencyName);
+
+      const dependencyEntry: IDependencyGraphEntry = ensureAndGetDependencyGraphEntry({
+        dependencyGraph,
+        packageName: dependencyName
+      });
+      dependencyEntry.dependents.push(project.packageName);
+    }
+  });
+}
+
+function processPackage({ dependencyGraph, project, rushConfiguration }: IProcessPackageOptions): void {
+  [
+    project.packageJson.dependencies,
+    project.packageJson.devDependencies,
+    project.packageJson.peerDependencies
+  ].forEach((dependencies) =>
+    addDependencies({
+      dependencies,
+      dependencyGraph,
+      project,
+      rushConfiguration
+    })
+  );
+}
+
+export function makeDependencyGraph(): IDependencyGraph {
+  const dependencyGraph: IDependencyGraph = {};
+  const rushConfiguration: RushConfiguration = RushConfiguration.loadFromDefaultLocation({
```

Comment:
You can hand the plugin instance the `RushConfiguration` instance in its constructor. It is accessible in the context of the action.

Reply:
Good call